mParticle-iterable/MPKitIterable.m

```diff
@@ -25,25 +33,19 @@ + (void)load {
 
 + (void)setCustomConfig:(IterableConfig *_Nullable)config {
     _customConfig = config;
-    _customUrlDelegate = config.urlDelegate;
 }
 
 - (BOOL)handleIterableURL:(NSURL *)url context:(IterableActionContext *)context {
     BOOL result = YES;
-    if (_customUrlDelegate == self) {
-        NSLog(@"mParticle -> Error: Iterable urlDelegate was set in custom config but points to the MKKitIterable instance. It should be a different object.");
-    } else if (_customUrlDelegate != nil && [((NSObject *)_customUrlDelegate) respondsToSelector:@selector(handleIterableURL:context:)]) {
-        result = [_customUrlDelegate handleIterableURL:url context: context];
-    } else if (_customUrlDelegate != nil) {
+    if (_customConfig.urlDelegate != nil && [((NSObject *)_customConfig.urlDelegate) respondsToSelector:@selector(handleIterableURL:context:)]) {
+        result = [_customConfig.urlDelegate handleIterableURL:url context: context];
+    } else if (_customConfig.urlDelegate != nil) {
         NSLog(@"mParticle -> Error: Iterable urlDelegate was set in custom config but didn't respond to the selector 'handleIterableURL:context:'");
     }
 
     NSString *destinationURL = url.absoluteString;
     NSDictionary *getAndTrackParams = nil;
-    NSString *clickedUrlString = _clickedURL.absoluteString;
-    if (clickedUrlString == nil) {
-        clickedUrlString = @"";
-    }
+    NSString *clickedUrlString = [NSString stringWithString:_clickedURL.absoluteString];
```

Comment:
Why remove the nil check which was added as a crash fix in the second most recent PR here: https://github.com/mparticle-integrations/mparticle-apple-integration-iterable/pull/8

The call to `[NSString stringWithString:_clickedURL.absoluteString];` will crash if `_clickedURL.absoluteString` is nil... Maybe it was misunderstood that `stringWithString` would return an empty string if the parameter is nil?